test/tests.py

```diff
@@ -0,0 +1,330 @@
+    succeed("systemctl stop electrs")
+
+
+@test("liquidd")
+def _():
+    assert_running("liquidd")
+    machine.wait_until_succeeds("elements-cli getnetworkinfo")
+    assert_matches("su operator -c 'elements-cli getnetworkinfo' | jq", '"version"')
+    succeed("su operator -c 'liquidswap-cli --help'")
+
+
+@test("clightning")
+def _():
+    assert_running("clightning")
+    assert_matches("su operator -c 'lightning-cli getinfo' | jq", '"id"')
+
+
+@test("lnd")
+def _():
+    assert_running("lnd")
+    assert_matches("su operator -c 'lncli getinfo' | jq", '"version"')
+    assert_no_failure("lnd")
+
+
+@test("lightning-loop")
+def _():
+    assert_running("lightning-loop")
+    assert_matches("su operator -c 'loop --version'", "version")
+    # Check that lightning-loop fails with the right error, making sure
+    # lightning-loop can connect to lnd
+    machine.wait_until_succeeds(
+        log_has_string(
+            "lightning-loop",
+            "Waiting for lnd to be fully synced to its chain backend, this might take a while",
+        )
+    )
+
+
+@test("btcpayserver")
+def _():
+    assert_running("nbxplorer")
+    machine.wait_until_succeeds(log_has_string("nbxplorer", "BTC: RPC connection successful"))
+    wait_for_open_port(ip("nbxplorer"), 24444)
+    assert_running("btcpayserver")
+    machine.wait_until_succeeds(log_has_string("btcpayserver", "Listening on"))
+    wait_for_open_port(ip("btcpayserver"), 23000)
+    # test lnd custom macaroon
+    assert_matches(
+        "sudo -u btcpayserver curl -s --cacert /secrets/lnd-cert "
+        '--header "Grpc-Metadata-macaroon: $(xxd -ps -u -c 1000 /run/lnd/btcpayserver.macaroon)" '
+        f"-X GET https://{ip('lnd')}:8080/v1/getinfo | jq",
+        '"version"',
+    )
+
+
+@test("spark-wallet")
+def _():
+    assert_running("spark-wallet")
+    wait_for_open_port(ip("spark-wallet"), 9737)
+    spark_auth = re.search("login=(.*)", succeed("cat /secrets/spark-wallet-login"))[1]
+    assert_matches(f"curl -s {spark_auth}@{ip('spark-wallet')}:9737", "Spark")
+
+
+@test("lightning-charge")
+def _():
+    assert_running("lightning-charge")
+    wait_for_open_port(ip("lightning-charge"), 9112)
+    machine.wait_until_succeeds(f"nc -z {ip('lightning-charge')} 9112")
+    charge_auth = re.search("API_TOKEN=(.*)", succeed("cat /secrets/lightning-charge-env"))[1]
+    assert_matches(
+        f"curl -s api-token:{charge_auth}@{ip('lightning-charge')}:9112/info | jq", '"id"'
+    )
+
+
+@test("nanopos")
+def _():
+    assert_running("nanopos")
+    wait_for_open_port(ip("nanopos"), 9116)
+    assert_matches(f"curl {ip('nanopos')}:9116", "tshirt")
+
+
+@test("joinmarket")
+def _():
+    assert_running("joinmarket")
+    machine.wait_until_succeeds(
+        log_has_string("joinmarket", "P2EPDaemonServerProtocolFactory starting on 27184")
+    )
+    machine.wait_until_succeeds(
+        log_has_string("joinmarket-yieldgenerator", "Failure to get blockheight",)
+    )
+
+
+@test("secure-node")
```

Comment:
Why did you place these tests under the label `secure-node`?

Reply:
Because the features tested here (`onion-chef`, `create-web-index`) are only available through `secure-node`.